### DISCUSSION TITLE
nan in loss

### DISCUSSION THREAD

**Discussion by mwitiderrick** [google/flax#2247] [Q&A] [answered]
Just trying Flax and I have nans in the loss. Here is the gist to the complete code running on Colab

https://gist.github.com/mwitiderrick/67f3482cecb3408dd14db374fc87a85a

What am I doing wrong?

**Comment by marcvanzee** [comment]
Hi @mwitiderrick! Did you try debugging this yourself already? If so, please provide some more information so we can narrow down the problem together. If not, I suggest looking at some tutorials on the web. For instance, Googling "why do i have nans in my loss" gave me the tutorial ["Common Causes of NANs During Training"](https://sisyphus.gitbook.io/project/deep-learning-basics/deep-learning-debug/common-causes-of-nans-during-training), which gives some suggestions for things you can try.

Please let us know how that is going!

**Comment by jheek** [comment, marked as answer]
You should apply `jax.nn.log_softmax` to normalize the logits before computing the cross entropy

**Comment by cgarciae** [comment]
You can use [optax.softmax_cross_entropy](https://optax.readthedocs.io/en/latest/api.html#optax.softmax_cross_entropy) to simplify your code a bit and get the correct loss.

**Comment by mwitiderrick** [comment]
The above suggestions fixed the issue, thanks.
